tests/controllers/excludename/matchname/chainsaw-test.yaml

```diff
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 Authors of Nimbus
+
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: match-names-add-csib
+spec:
+  description: >
+    1. Adds a csib for matchNames with dev, staging
+    2. Delete staging ns
+    3. Update csib for matchNames with dev, prod
+  steps:
+    - name: "Create prod, dev, staging Namespaces"
+      try:
+        - apply:
+            file: ns.yaml
+
+    - name: "Create a SecurityIntent"
+      try:
+        - apply:
+            file: ../../resources/namespaced/escape-to-host-si.yaml
+
+    - name: "Create a ClusterSecurityIntentBinding"
+      try:
+        - apply:
+            file: escape-to-host-csib.yaml
+
+    - name: "Verify ClusterNimbusPolicy creation"
+      try:
+        - assert:
+            file: cluster-nimbus-policy-assert.yaml
+
+    - name: "Verify NimbusPolicy creation in dev namespace"
+      try:
+        - assert:
+            file: nimbus-policy-assert-dev.yaml
+
+    - name: "Verify NimbusPolicy creation in staging namespace"
+      try:
+        - assert:
+            file: nimbus-policy-assert-staging.yaml
+
+    - name: "Verify NimbusPolicy absence in prod"
+      try:
+        - script:
+            content: kubectl -n prod get np nimbus-ctrl-gen-escape-to-host
+            check:
+              ($error != null): true
+
+    - name: "Delete staging namespace"
+      try:
+        - delete:
+            ref:
+              apiVersion: v1
+              kind: namespace
+              name: staging
+            expect:
+              - match:
+                  apiVersion: v1
+                  kind: namespace
+                  name: staging
+                check:
+                  ($error != null): true
+
+    - name: "Verify the NimbusPolicy deletion in staging"
+      try:
+        - script:
+            content: kubectl -n staging get np nimbus-ctrl-gen-escape-to-host
+            check:
+              ($error != null): true
+
```

Comment:
This test step can be removed. Deleting a namespace inherently deletes all its child resources.